storage/storage_dest.go

```diff
@@ -783,11 +778,15 @@ func (s *storageImageDestination) createNewLayer(index int, layerDigest digest.D
 		options := &graphdriver.ApplyDiffWithDifferOpts{
 			Flags: flags,
 		}
-		if err := s.imageRef.transport.store.ApplyDiffFromStagingDirectory(layer.ID, diffOutput.Target, diffOutput, options); err != nil {
-			_ = s.imageRef.transport.store.Delete(layer.ID)
-			return nil, err
+
+		args := storage.ApplyStagedLayerOptions{
+			ID:          newLayerID,
+			ParentLayer: parentLayer,
+
+			DiffOutput:  diffOutput,
+			DiffOptions: options,
 		}
-		return layer, nil
+		return s.imageRef.transport.store.ApplyStagedLayer(args)
```

Comment:
This is pre-existing, but here we need to silently succeed on a `ErrDuplicateID`, similarly to `PutLayer` below.

Reply:
thanks, amended now